src/util/types/Util.d.ts

```diff
@@ -1,2 +1,44 @@
 export type RequiredExcept<T, K extends keyof T> = Partial<Pick<T, K>> & Required<Omit<T, K>>;
 export type PartialRequired<T, K extends keyof T> = Partial<Omit<T, K>> & Required<Pick<T, K>>;
+
+export const enum ClientEvents {
+	ChannelCreate = 'channelCreate',
+	ChannelDelete = 'channelDelete',
+	ChannelPinsUpdate = 'channelPinsUpdate',
+	ChannelUpdate = 'channelUpdate',
+	Debug = 'debug',
+	Error = 'error',
+	GuildBanAdd = 'guildBanAdd',
+	GuildBanRemove = 'guildBanRemove',
+	GuildCreate = 'guildCreate',
+	GuildDelete = 'guildDelete',
+	GuildEmojisUpdate = 'guildEmojisUpdate',
+	GuildIntegrationsUpdate = 'guildIntegrationsUpdate',
+	GuildMemberAdd = 'guildMemberAdd',
+	GuildMemberRemove = 'guildMemberRemove',
+	GuildMembersChunk = 'guildMembersChunk',
+	GuildMemberUpdate = 'guildMemberUpdate',
+	GuildRoleCreate = 'guildRoleCreate',
+	GuildRoleDelete = 'guildRoleDelete',
+	GuildRoleUpdate = 'guildRoleUpdate',
+	GuildUpdate = 'guildUpdate',
+	InviteCreate = 'inviteCreate',
+	InviteDelete = 'inviteDelete',
+	MessageCreate = 'messageCreate',
+	MessageDelete = 'messageDelete',
+	MessageDeleteBulk = 'messageDeleteBulk',
+	MessageReactionAdd = 'messageReactionAdd',
+	MessageReactionRemove = 'messageReactionRemove',
+	MessageReactionRemoveAll = 'messageReactionRemoveAll',
+	MessageUpdate = 'messageUpdate',
+	PresenceUpdate = 'presenceUpdate',
+	Ready = 'ready',
+	Resumed = 'resumed',
+	ShardOnline = 'shardOnline',
+	TypingStart = 'typingStart',
+	UserUpdate = 'userUpdate',
+	VoiceServerUpdate = 'voiceServerUpdate',
+	VoiceStateUpdate = 'voiceStateUpdate',
+	WebhooksUpdate = 'webhooksUpdate',
+	WSDebug = 'wsDebug'
```

Comment:
I'm still split on this being here when REST uses the internal REST event enum; but I can close an eye